packages/react/src/misc/adapters/InviteWalletsGateway.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  InviteData,
+  InviteDocument,
+  InviteVariables,
+  SafeApolloClient,
+} from '@lens-protocol/api-bindings';
+import {
+  IInviteWalletsGateway,
+  InviteWalletsRequest,
+  WalletAlreadyInvitedError,
+} from '@lens-protocol/domain/use-cases/wallets';
+import { PromiseResult, assertError, failure, success } from '@lens-protocol/shared-kernel';
+
+export class InviteWalletsGateway implements IInviteWalletsGateway {
+  constructor(private apolloClient: SafeApolloClient) {}
+
+  async invite(request: InviteWalletsRequest): PromiseResult<void, WalletAlreadyInvitedError> {
+    try {
+      await this.apolloClient.mutate<InviteData, InviteVariables>({
+        mutation: InviteDocument,
+        variables: {
+          request: {
+            invites: request.wallets,
+          },
+        },
+      });
+
+      return success();
+    } catch (error) {
```

Comment:
we don't really use this pattern but I don't see how else to catch API errors and turn them into user meaningful errors.